src/models/PrimaryAuth.js

```diff
@@ -91,14 +91,18 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
         if (deviceFingerprintEnabled) {
           authClient.options.headers['X-Device-Fingerprint'] = this.get('deviceFingerprint');
         }
-        return authClient.signIn({
+        var signInArgs = {
           username: username,
           password: password,
           options: {
             warnBeforePasswordExpired: true,
             multiOptionalFactorEnroll: multiOptionalFactorEnroll
           }
-        })
+        };
+        if (this.appState.get('unauthenticatedToken')) {
+          signInArgs.stateToken = this.appState.get('unauthenticatedToken');
+        }
```

Comment:
if `this.appState.get('unauthenticatedToken')` is null, `signInArgs.stateToken` will be undefined. As long as the API handles this, it should be fine :) Maybe write a test for this case to ensure it works.